### PR TITLE
Implement API to add, delete and list available secrets

### PR DESCRIPTION
**Pull request by philippjfr**
Implements https://github.com/Anaconda-Platform/ae5-tools/issues/61

`AE5UserSession.secret_add`: Allows adding (and updating) a secret by providing `key` and `value`
`AE5UserSession.secret_delete`: Allows deleting a secret by providing `key`
`AE5UserSession.secret_list`: Returns a list of secrets declared for the user (but not their values)

### CODE REVIEW

**Comment by jlstevens** [comment]
First quick comment: if you are manipulating one secret at a time shouldn't it be `AE5UserSession.secret_add`, `AE5UserSession.secret_delete` and `AE5UserSession.secret_list` (singular)?

Plural makes sense if it were to support changing multiple at once.

**Comment by philippjfr** [comment]
The main consistency now is that `secret_list` returns a simple list not a set of records. I'll leave it up to @mcg1969 whether that's sensible or I should somehow wrap up the response to emulate records.

File: ae5_tools/cli/commands/secrets.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+from ..login import cluster_call
+from ..utils import global_options, yes_option
+
+
+@click.group(short_help='list, add, delete',
+             epilog='Type "ae5 secret <command> --help" for help on a specific command.')
+@global_options
+def secret():
+    '''Commands related to static endpoints.'''
```
Comment (philippjfr) [inline ae5_tools/cli/commands/secrets.py:11]:
```suggestion
    '''Commands related to managing user secrets.'''
```

**Comment by jlstevens** [comment]
I've just had a go testing `add`, `delete` and `list` and other than taking a minute or so to update, it seems to be working fine!

**Comment by philippjfr** [comment]
>other than taking a minute or so to update

To be clear about this, it's the secrets files in running sessions located in `/var/run/secrets/user_credentials/` that take a minute to update, both the UI and the list command (which are both driven by the REST API) respond to add/delete actions immediately.

**Comment by philippjfr** [comment]
Pretty sure the test failures are unrelated to this PR.

**Comment by joshburt** [comment]
* I've left the command as `add` and updated the documentation to explicitly denote that it is an add or update.
* I left the formatting support in place.
After writing the tests cased for it I found the interaction to be a bit "bulky":

```json
[
  {
    "secrets": [
      "AE5_HOSTNAME",
      "AE5_PASSWORD",
      "AE5_USERNAME",
      "MLFLOW_REGISTRY_URI",
      "MLFLOW_TRACKING_TOKEN",
      "MLFLOW_TRACKING_URI"
    ]
  }
]
```
It conforms to the other commands, however programatic access looks something like:
```python
my_secrets = sercret_list()[0]["secrets"]
```
* As designed we can not retrieve the value of a secret once created through the tools.  The test coverage I've added does not check the value stored or updated in the tests.  If we want this I put together a fixture for a project which will expose secrets within a REST endpoint.  I ultimately did not include it as it was a lot of code to validate this.  If we feel this is critical let me know and I'll extend the system tests to include this fixture and the setup/tear down of the project during the tests.

**Comment by joshburt** [comment]
Output format updated.  Looks much better pivoted this way.  I've landed on a column name of `secret_name`.  Let me know if we want to use a different one.

> python -m ae5_tools.cli.main secret list
```
            secret_name
------------------------------------
123-123
22624586-83db-42d3-93d7-2068b84c82c1
2ce2ff39_4599_4061_951d_987ea562ccd3
36b7120d_df60_4f3a_9b89_033fde3d18cc
377c578e_7149_46f0_8a36_589f526015ac
41048992-fb8f-43b4-9827-4084c893fac1
95f0d467_4dbb_4fa0_962b_509f4d020d1b
95fa06b2-8f14-4477-bc35-9b48589de842
AE5_HOSTNAME
AE5_PASSWORD
AE5_USERNAME
MLFLOW_REGISTRY_URI
MLFLOW_TRACKING_TOKEN
MLFLOW_TRACKING_URI
cd3de8bb-6efd-4a44-91cd-6a193940521c
f15db56f-f5ce-430b-8d08-dc3733f46242
ff81dd8f_4948_4b39_a343_487d5ca15b54

```

> python -m ae5_tools.cli.main secret list --format json
```
[
  {
    "secret_name": "123-123"
  },
  {
    "secret_name": "22624586-83db-42d3-93d7-2068b84c82c1"
  },
  {
    "secret_name": "2ce2ff39_4599_4061_951d_987ea562ccd3"
  },
  {
    "secret_name": "36b7120d_df60_4f3a_9b89_033fde3d18cc"
  },
  {
    "secret_name": "377c578e_7149_46f0_8a36_589f526015ac"
  },
  {
    "secret_name": "41048992-fb8f-43b4-9827-4084c893fac1"
  },
  {
    "secret_name": "95f0d467_4dbb_4fa0_962b_509f4d020d1b"
  },
  {
    "secret_name": "95fa06b2-8f14-4477-bc35-9b48589de842"
  },
  {
    "secret_name": "AE5_HOSTNAME"
  },
  {
    "secret_name": "AE5_PASSWORD"
  },
  {
    "secret_name": "AE5_USERNAME"
  },
  {
    "secret_name": "MLFLOW_REGISTRY_URI"
  },
  {
    "secret_name": "MLFLOW_TRACKING_TOKEN"
  },
  {
    "secret_name": "MLFLOW_TRACKING_URI"
  },
  {
    "secret_name": "cd3de8bb-6efd-4a44-91cd-6a193940521c"
  },
  {
    "secret_name": "f15db56f-f5ce-430b-8d08-dc3733f46242"
  },
  {
    "secret_name": "ff81dd8f_4948_4b39_a343_487d5ca15b54"
  }
]
```

> python -m ae5_tools.cli.main secret list --format csv
```
secret_name
123-123
22624586-83db-42d3-93d7-2068b84c82c1
2ce2ff39_4599_4061_951d_987ea562ccd3
36b7120d_df60_4f3a_9b89_033fde3d18cc
377c578e_7149_46f0_8a36_589f526015ac
41048992-fb8f-43b4-9827-4084c893fac1
95f0d467_4dbb_4fa0_962b_509f4d020d1b
95fa06b2-8f14-4477-bc35-9b48589de842
AE5_HOSTNAME
AE5_PASSWORD
AE5_USERNAME
MLFLOW_REGISTRY_URI
MLFLOW_TRACKING_TOKEN
MLFLOW_TRACKING_URI
cd3de8bb-6efd-4a44-91cd-6a193940521c
f15db56f-f5ce-430b-8d08-dc3733f46242
ff81dd8f_4948_4b39_a343_487d5ca15b54
```

**Comment by jlstevens** [comment]
Looks much better to me!

**Comment by joshburt** [comment]
> 

Confirmation on the output formatting was the most important part.  Thanks @jlstevens !

**Pull request merged**